ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java

```diff
@@ -0,0 +1,782 @@
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({
+        BaseListFragment.class,
+        DefaultStateViewHelper.class,
+        BaseListFragmentTest.TestBaseListFragment.class,
+        Looper.class
+})
+public class BaseListFragmentTest {
+
+    private TestBaseListFragment testBaseListFragment;
+    private Boolean isMethodSuccessfullyRun;
+
+    @Mock
+    private Looper looper;
+    @Mock
+    private View mockMRoot, mockView, mockCreateView;
+    @Mock
+    private RecyclerView mockMRecyclerView;
+    @Mock
+    private LayoutInflater mockInflater;
+    @Mock
+    private ViewGroup mockContainer;
+    @Mock
+    private Bundle savedInstanceState;
+    @Mock
+    private LinearLayoutManager mockLinearLayoutManager;
+    @Mock
+    private DefaultStateViewHelper mockMDefaultStateViewHelper;
+    @Mock
+    private CustomStateViewHelper mockCustomStateViewHelper;
+    @Mock
+    private DefaultStateViewHelper mockDefaultStateViewHelper;
+    @Mock
+    private OnListPageStateChangeListener mockListPageChangeStateListener;
+    @Mock
+    private Context mockContext;
+    @Mock
+    private EndlessRecyclerViewScrollAdapter mockAdapter;
+    @Mock
+    private Handler handler;
+    @Captor
+    private ArgumentCaptor<Runnable> runnableCaptor;
+    @Mock
+    private RecyclerView.OnScrollListener onScrollListener;
+    @Mock
+    private EndlessRecyclerViewScrollAdapter.OnLoadMoreListener loadMoreListener;
+    @Mock
+    private EndlessRecyclerViewScrollListener mloadMoreListener;
+
+    @Before
+    public void setUp() {
+        isMethodSuccessfullyRun = Boolean.FALSE;
+        MockitoAnnotations.initMocks(this);
+        testBaseListFragment = new TestBaseListFragment();
+        Whitebox.setInternalState(testBaseListFragment, "mCustomStateViewHelper", mockCustomStateViewHelper);
+        Whitebox.setInternalState(testBaseListFragment, "mDefaultStateViewHelper", mockDefaultStateViewHelper);
+        Whitebox.setInternalState(testBaseListFragment, "mListPageChangeStateListener", mockListPageChangeStateListener);
+        PowerMockito.mockStatic(Looper.class);
+        when(Looper.getMainLooper()).thenReturn(looper);
+    }
+
+    @Test
+    public void onCreateViewTest() {
+        //Arrange
+        BaseListFragment mockBaseListFragment = Mockito.mock(BaseListFragment.class);
+        Mockito.doCallRealMethod().when(mockBaseListFragment).onCreateView(mockInflater, mockContainer, savedInstanceState);
+        Mockito.doReturn(mockCreateView).when(mockInflater).inflate(R.layout.ko__fragment_list, mockContainer, false);
+        Mockito.doReturn(mockMRecyclerView).when(mockCreateView).findViewById(R.id.ko__list);
+        PowerMockito.suppress(methods(BaseListFragment.class, "setupView"));
+        //Act
+        View view  = mockBaseListFragment.onCreateView(mockInflater, mockContainer, savedInstanceState);
+        //Assert
+        Assert.assertNotNull(view);
+    }
+
+    class TestBaseListFragment extends BaseListFragment {
```

Comment:
❗️ This file contains multiple top-level classes

This allows multiple definitions for a class. Which definition gets used is affected by the order in which the source files are passed to the compiler. Having only one top-level class per file guarantees that the class files generated by compilation, and the behavior of the resulting program, are independent of the order in which the source files are passed to the compiler. So Consider the proposed solution.

## The possible solution this case

Consider splitting the current file in two files:

- BaseListFragmentTest.java - containing the class `BaseListFragmentTest`
- TestBaseListFragment.java - containin the class `TestBaseListFragment` 

<a id="c10a6790-50d9-11e8-89e2-ed64d7ac7be5" href="none" codereview-uuid-reserved-tag />

Reply:
done